packages/block-editor/src/components/provider/use-block-entity-sync.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * External dependencies
+ */
+import { last, noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import isShallowEqual from '@wordpress/is-shallow-equal';
+import { useEffect, useRef } from '@wordpress/element';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+export function useBlockEntitySync( {
+	registry,
+	value: controlledBlocks,
```

Comment:
Why do I need  this here? can't we retrieve it from the "clientId" and the block-editor store?

Reply:
This is where we pass the post's blocks from `core/editor`. We can't retrieve them without making the `block-editor` depend on `core/editor`.

Reply:
really, isn't it the same thing as `getBlocks` with no rootclientId?

Reply:
I'm asking the questions to confront the "theory" in my head with the reality in this PR :) the theory could definitely be wrong though :P 

for me

`value = getBlocks( clientId )`  and clientId is nullable (for the root) and it's equal to the container block's id for the InnerBlocks.

Reply:
oohhhhhhh, sorry, I get it now. this is  the  prop directly, so it's the thing that needs to be synced. theory definitely was wrong here.

Reply:
yep! Maybe some comments would help with this? I was hoping that `controlledBlocks` could make it more obvious :D